setup/bundles.rst

```diff
@@ -198,6 +198,56 @@ adding a ``setDefined()`` method. The recommended check in this case would be::
     to detect the version of the ``symfony/http-kernel`` component, because it
     is the component where this constant is defined.
 
+Generic conditions for component version detection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You should strive to detect the exact feature that you want to use as
+soon as it is available. That is not always possible however, so here is
+a list of generic conditions you may use for lack of a better
+alternative. These conditions detect new APIs as soon as they are
+available.
+
+2.6
+...
+
+OptionsResolver
+"""""""""""""""
+
+.. code-block:: php
+
+    method_exists(Symfony\Component\OptionsResolver\OptionsResolver, 'setDefined')
+
+2.7
+...
+
+Form
+""""
+
+.. code-block:: php
+
+    method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')
+
+2.8
+...
+
+Form
+""""
+
+.. code-block:: php
+
+    method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
+
+3.0
+...
+
+Form
```

Comment:
All their API? I don't think so. The goal is to detect what version a component is in, not to list all the changes. The problem I'm trying to fix is that sometimes you just can't easily detect a feature, like for instance if the `choices_as_values` option is available or not. So instead, I propose to give conditions that could help detect the version and thus if the option is available or not.

> this should be done for every component as well,

I'm not event sure it's doable for every version of every component.


> It should be done on a case by case basis, so leading the way as this was done before should be enough.

I don't understand what you mean by that exactly. Can you please explain?